File: 0035-search-insert-position/Solution.cs
public class Solution
{
    public int SearchInsert(int[] nums, int target)
    {
        var leftIndex = 0;
        var rightIndex = nums.Length - 1;
        while (leftIndex <= rightIndex)
        {
            var middleIndex = (leftIndex + rightIndex) / 2;
            if (nums[middleIndex] == target)
            {
                return middleIndex;
            }
            if (nums[middleIndex] < target)
            {
                leftIndex = middleIndex + 1;
            }
            else
            {
                rightIndex = middleIndex - 1;
            }
        }
        return leftIndex;
    }
}

File: 0035-search-insert-position/Solution.go
package Leetcode

func searchInsert(nums []int, target int) int {
	leftIndex := 0
	rightIndex := len(nums) - 1
	for leftIndex <= rightIndex {
		midIndex := (leftIndex + rightIndex) / 2
		if nums[midIndex] == target {
			return midIndex
		} else if nums[midIndex] < target {
			leftIndex = midIndex + 1
		} else {
			rightIndex = midIndex - 1
		}
	}
	return leftIndex
}
